File: java/src/Behavioral/Visitor/BookInfo.java
package Behavioral.Visitor;


//BookInfo.java - one of three concrete Visitees

public class BookInfo extends AbstractTitleInfo {  
   private String author;
    
   public BookInfo(String titleName, String author) {
       this.setTitleName(titleName);
       this.setAuthor(author);
   }    
   
   public void setAuthor(String authorIn) {
       this.author = authorIn;
   }
   public String getAuthor() {
       return this.author;
   }
   
   public void accept(TitleBlurbVisitor titleBlurbVisitor) {
       titleBlurbVisitor.visit(this);
   }
}

File: csharp/src/Behavioral/Visitor/BookInfo.cs
//BookInfo.java - one of three concrete Visitees

namespace Behavioral.Visitor {


public class BookInfo : AbstractTitleInfo {  
   private string author;
    
   public BookInfo(string titleName, string author) {
       this.setTitleName(titleName);
       this.setAuthor(author);
   }    
   
   public void setAuthor(string authorIn) {
       this.author = authorIn;
   }
   public string getAuthor() {
       return this.author;
   }
   
   override public void accept(TitleBlurbVisitor titleBlurbVisitor) {
       titleBlurbVisitor.visit(this);
   }
}

}
